packages/arb-avm-cpp/avm/src/machinestate/machinestate.cpp

```diff
@@ -38,11 +38,10 @@ MachineState::MachineState()
       inbox(pool.get()) {}
 
 MachineState::MachineState(const std::vector<CodePoint>& code_,
-                           const value& static_val_)
-    : pool(std::make_unique<TuplePool>()),
-      pendingInbox(pool.get()),
-      context({0, 0}),
-      inbox(pool.get()) {
+                           const value& static_val_,
+                           std::shared_ptr<TuplePool>& pool_)
+    : pendingInbox(pool_.get()), context({0, 0}), inbox(pool_.get()) {
+    pool = pool_;
```

Comment:
After changing `pool_` to pass by value, set `pool(std::move(pool_))` in the initializer list